pkg/cmd/ssh/options.go

```diff
@@ -0,0 +1,228 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ssh
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/cmd/base"
+	"github.com/gardener/gardener/pkg/utils"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh"
+	cryptossh "golang.org/x/crypto/ssh"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+// Options is a struct to support target command
+type Options struct {
+	base.Options
+
+	// Interactive can be used to toggle between gardenctl just
+	// providing the bastion host while keeping it alive (non-interactive),
+	// or gardenctl opening the SSH connection itself (interactive). For
+	// interactive mode, a NodeName must be specified as well.
+	Interactive bool
+
+	// NodeName is the name of the Shoot cluster node that the user wants to
+	// connect to. If this is left empty, grdenctl will only establish the
+	// bastion host, but leave it up to the user to SSH themselves.
+	NodeName string
+
+	// CIDRs is a list of IP address ranges to allowed for accessing the
+	// created Bastion host. If not given, gardenctl will attempt to
+	// auto-detect the user's IP and allow only it (i.e. use a /32 netmask).
+	CIDRs []string
+
+	// SSHPublicKeyFile is the full path to the file containing the user's
+	// public SSH key. If not given, gardenctl will create a new temporary keypair.
+	SSHPublicKeyFile string
+
+	// SSHPrivateKeyFile is the full path to the file containing the user's
+	// private SSH key. This is only set if no key was given and a temporary keypair
+	// was generated. Otherwise gardenctl relies on the user's SSH agent.
+	SSHPrivateKeyFile string
+
+	// WaitTimeout is the maximum time to wait for a bastion to become ready.
+	WaitTimeout time.Duration
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		Options: base.Options{
+			IOStreams: ioStreams,
+		},
+		Interactive: true,
+		WaitTimeout: 10 * time.Minute,
+	}
+}
+
+// Complete adapts from the command line args to the data required.
+func (o *Options) Complete(f util.Factory, cmd *cobra.Command, args []string, stdout io.Writer) error {
+	if len(o.CIDRs) == 0 {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		publicIP, err := f.PublicIP(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to determine your system's public IP address: %w", err)
```

Comment:
the help text of the cidr flag is

> CIDRs to allow access to the bastion host; if not given, the host's public IP is auto-detected.

To better relate the `cidr` flag to the error message either change the help text to
`CIDRs to allow access to the bastion host; if not given, your system's public IP is auto-detected.`

or adapt this error message accordingly and use "host" instead of "system"